altair/vegalite/v5/api.py

```diff
@@ -380,24 +422,625 @@ def check_fields_and_encodings(parameter: Parameter, field_name: str) -> bool:
+    **constraints: Any,
+) -> _ConditionType:
+    composed: _PredicateType
+    if _is_undefined(predicate):
+        if more_predicates or constraints:
+            composed = _parse_when_compose(more_predicates, constraints)
+        else:
+            msg = (
+                f"At least one predicate or constraint must be provided, "
+                f"but got: {predicate=}"
+            )
+            raise TypeError(msg)
+    elif more_predicates or constraints:
+        predicates = predicate, *more_predicates
+        composed = _parse_when_compose(predicates, constraints)
+    else:
+        composed = predicate
+    return _predicate_to_condition(composed, empty=empty)
+
+
+def _parse_literal(val: Any, *, str_as_lit: bool) -> dict[str, Any]:
+    if isinstance(val, str) and not str_as_lit:
+        return utils.parse_shorthand(val)
+    elif _is_one_or_seq_literal_value(val):
+        return value(val)
+    else:
+        msg = f"Expected one or more literal values, but got: {type(val).__name__!r}"
+        raise TypeError(msg)
+
+
+def _parse_then(
+    statement: _StatementOrLiteralType,
+    kwargs: dict[str, Any],
+    *,
+    lit: bool,
+) -> dict[str, Any]:
+    if isinstance(statement, core.SchemaBase):
+        statement = statement.to_dict()
+    elif not isinstance(statement, dict):
+        statement = _parse_literal(statement, str_as_lit=lit)
+    statement.update(kwargs)
+    return statement
+
+
+def _parse_otherwise(
+    statement: _StatementOrLiteralType,
+    conditions: _Conditions,
+    kwargs: dict[str, Any],
+    *,
+    lit: bool,
+) -> SchemaBase | _Conditions:
+    selection: SchemaBase | _Conditions
+    if isinstance(statement, core.SchemaBase):
+        selection = statement.copy()
+        conditions.update(**kwargs)  # type: ignore[call-arg]
+        selection.condition = conditions["condition"]
+    else:
+        if not isinstance(statement, dict):
+            statement = _parse_literal(statement, str_as_lit=lit)
+        selection = conditions
+        selection.update(**statement, **kwargs)  # type: ignore[call-arg]
+    return selection
+
+
+class _BaseWhen(Protocol):
+    # NOTE: Temporary solution to non-SchemaBase copy
+    _condition: _ConditionType
+
+    def _when_then(
+        self,
+        statement: _StatementOrLiteralType,
+        kwargs: dict[str, Any],
+        *,
+        lit: bool,
+    ) -> _ConditionType:
+        condition = _deepcopy(self._condition)
+        then = _parse_then(statement, kwargs, lit=lit)
+        condition.update(then)
+        return condition
+
+
+class When(_BaseWhen):
+    """Utility class for ``when-then-otherwise`` conditions.
+
+    Represents the state after calling :func:`.when()`.
+
+    This partial state requires calling :meth:`When.then()` to finish the condition.
+
+    References
+    ----------
+    `polars.expr.whenthen <https://github.com/pola-rs/polars/blob/b85c5e0502ca99c77742ee25ba177e6cd11cf100/py-polars/polars/expr/whenthen.py>`__
```

Comment:
Marking this for discussion of all `polars` references. 

I feel there is a need to reference `polars`, but I'm not sure the "best" way to do so. 
There is [intersphinx](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html), but I think that might only solve for the user guide/ api reference links.